apps/site/assets/ts/__v3api.d.ts

```diff
@@ -17,6 +19,17 @@ export interface Headsign {
   train_number: string | null;
 }
 
+export interface HeadsignWithCrowding {
+  name: string;
+  time_data_with_crowding_list: PredictedOrScheduledTimeWithCrowding[];
+  train_number: string | null;
+}
+
+export interface PredictedOrScheduledTimeWithCrowding {
+  time_data: PredictedOrScheduledTime;
+  crowding: CrowdingType;
+}
```

Comment:
Does this have an advantage over the following?

```suggestion
export interface PredictedOrScheduledTimeWithCrowding extends PredictedOrScheduledTime {
  crowding: CrowdingType;
}
```

Reply:
This continues what I was rambling about above, but in my mind the advantage is in better modeling the data. That is to say, `crowding` isn't a property of `predicted time`. Those are both properties of the vehicle/trip approaching this stop. Especially on the server side of this work, it felt to me like a big part of what made the existing functions confusing is that they weren't operating on data that was structured in a way that communicated to me. Different sorts of data were combined and shifted around over and over in a way where I had a very hard time understanding the intentions.